bin/lib/invalidUsernames.js

```diff
@@ -0,0 +1,160 @@
+      }
+
+      const invalidUsernames = await getInvalidUsernames(config)
+      const host = SolidHost.from({ port: config.port, serverUri: config.serverUri })
+      const accountManager = getAccountManager(config, host)
+
+      if (options.notify) {
+        return notifyUsers(invalidUsernames, accountManager, config)
+      }
+
+      if (options.delete) {
+        return deleteUsers(invalidUsernames, accountManager, config, host)
+      }
+
+      listUsernames(listUsernames)
+    })
+}
+
+async function createNewIndexFile (username, accountManager, invalidUsernameTemplate, dateOfRemoval, supportEmail, fileOptions) {
+  const userDirectory = accountManager.accountDirFor(username)
+  const currentIndex = path.join(userDirectory, 'index.html')
+  const currentIndexExists = await fileExists(currentIndex)
+  const backupIndex = path.join(userDirectory, 'index.backup.html')
+  const backupIndexExists = await fileExists(backupIndex)
+  if (currentIndexExists && !backupIndexExists) {
+    await fileRename(currentIndex, backupIndex)
+    const newIndexSource = invalidUsernameTemplate({
+      username,
+      dateOfRemoval,
+      supportEmail
+    })
+    fs.writeFileSync(currentIndex, newIndexSource, fileOptions)
+    console.info(`index.html updated for user ${username}`)
+  }
+}
+
+async function deleteUsers (usernames, accountManager, config, host) {
+  const oidcManager = fromServerConfig({
+    ...config,
+    host
+  })
+  const deletingUsers = usernames
+    .map(async username => {
+      try {
+        const user = accountManager.userAccountFrom({ username })
+        await oidcManager.users.deleteUser(user)
+      } catch (error) {
+        if (error.message !== 'No email given') {
+          // 'No email given' is an expected error that we want to ignore
+          throw error
+        }
+      }
+      const userDirectory = accountManager.accountDirFor(username)
+      await fs.remove(userDirectory)
+    })
+  await Promise.all(deletingUsers)
+  console.info(`Deleted ${deletingUsers.length} users succeeded`)
+}
+
+function getAccountManager (config, host) {
+  const ldp = new LDP(config)
+  return AccountManager.from({
+    host,
+    store: ldp,
+    multiuser: config.multiuser
+  })
+}
+
+async function getInvalidUsernames (config) {
+  const files = await util.promisify(fs.readdir)(config.root)
+  const hostname = new URL(config.serverUri).hostname
+  const isUserDirectory = new RegExp(`.${hostname}$`)
+  return files
+    .filter(file => isUserDirectory.test(file))
+    .map(userDirectory => userDirectory.substr(0, userDirectory.length - hostname.length - 1))
+    .filter(username => !isValidUsername(username) || !blacklistService.validate(username))
+}
+
+function listUsernames (usernames) {
+  if (usernames.length === 0) {
+    console.info('No invalid usernames was found')
+  }
+  console.info(`${usernames.length} invalid usernames were found:${usernames.map(username => `\n- ${username}`)}`)
```

Comment:
Shouldn't that go into an `else` branch?

Reply:
Oh, forgot a return in the if-block (that's at least how I like to write that logic).